### PR TITLE
[OBSDEF-7255] ability to pass storageclass for atlas

### PR DESCRIPTION
**Pull request by sumithreddy**
## Purpose
[OBSDEF-7255](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-7255)
configurable storageclass for atlas deployments

## PR checklist
- [x] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [x] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
_Paste URL of charts-custom-ci job here_
https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/293/
https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/294/

_Paste the output of your helm install/vSphere7 deployment testing here_

**make test**
<pre>
Charts:      30 passed, 30 total
Test Suites: 49 passed, 49 total
Tests:       238 passed, 238 total
Snapshot:    0 passed, 0 total
Time:        58.801053958s
</pre>

**make build**
<pre><font color="#D3D7CF"><b>Update Complete. ⎈Happy Helming!⎈</b></font>
<font color="#D3D7CF"><b>Saving 1 charts</b></font>
<font color="#D3D7CF"><b>Deleting outdated charts</b></font>
<font color="#D3D7CF"><b>Successfully packaged chart and saved it to: docs/objectscale-vsphere-0.79.0-1244.tgz</b></font>
</pre>

**helm install**
<pre><font color="#3465A4">[</font><font color="#CC0000">root@flexdevkit:~</font><font color="#3465A4">]</font><font color="#4E9A06">$ </font><font color="#D3D7CF"><b>kubectl get pods</b></font>
<font color="#D3D7CF"><b>NAME                                               READY   STATUS    RESTARTS   AGE</b></font>
<font color="#D3D7CF"><b>atlas-operator-9c8bbc8f8-8ksdm                     1/1     Running   0          7m44s</b></font>
<font color="#D3D7CF"><b>ecs-manager-bookkeeper-operator-7ddf8c955b-n2thn   1/1     Running   0          7m45s</b></font>
<font color="#D3D7CF"><b>ecs-manager-pravega-operator-dd85d64f6-jwddw       1/1     Running   0          7m45s</b></font>
<font color="#D3D7CF"><b>ecs-manager-service-pod-fcbcc69fb-hrptg            1/1     Running   0          7m45s</b></font>
<font color="#D3D7CF"><b>objectscale-dcm-847b46fb4d-w82nr                   1/1     Running   0          7m45s</b></font>
<font color="#D3D7CF"><b>objectscale-dcm-atlas-0                            1/1     Running   0          7m21s</b></font>
<font color="#D3D7CF"><b>objectscale-federation-54fcddf9bc-58k8p            1/1     Running   0          7m45s</b></font>
<font color="#D3D7CF"><b>objectscale-federation-atlas-0                     1/1     Running   0          7m21s</b></font>
<font color="#D3D7CF"><b>objectscale-gateway-6bcf585ddc-q2dmq               1/1     Running   5          7m45s</b></font>
<font color="#D3D7CF"><b>objectscale-iam-596c6df4b4-hxmdx                   1/1     Running   3          7m45s</b></font>
<font color="#D3D7CF"><b>objectscale-iam-atlas-0                            1/1     Running   0          7m21s</b></font>
<font color="#D3D7CF"><b>objectscale-operator-5778c84c99-z7cgq              2/2     Running   0          7m45s</b></font>
<font color="#D3D7CF"><b>zookeeper-operator-d557fd46d-xn5pj                 1/1     Running   0          7m45s</b></font>
<font color="#3465A4">[</font><font color="#CC0000">root@flexdevkit:~</font><font color="#3465A4">]</font><font color="#4E9A06">$ </font>
</pre>

### CODE REVIEW

**Review by skripg — APPROVED** [review]

File: objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml [hunk 1]
```diff
@@ -14,9 +14,7 @@ spec:
     pullPolicy: {{ .Values.atlas.image.pullPolicy }}
   persistence:
     spec:
-      {{- if .Values.global.storageClassName}}
-      storageClassName: {{.Values.global.storageClassName}}
-      {{- end }}
+      storageClassName: {{ default .Values.global.storageClassName .Values.atlas.persistence.storageClassName }}
```
Comment (eddiepavkovic) [inline objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:17]: so if neither one is set this is blank and i think it picks up the default storage class or fails.   did we test this?

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]
please run custom-ci job and post the result before merging.

File: objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml (hunk 1 above)
Comment (sumithreddy) [reply objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:17]: I am running deployments now for various scenarios.

https://github.com/EMCECS/charts/blob/4c7f0a0878a2179aa5c1c0ea0cd34399eb52c804/ecs-cluster/templates/ecs-cluster.yaml#L79 -> worked fine with no issues till now. Need to check if omitEmpty set somewhere

Comment (sumithreddy) [reply objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:17]: storageClass and registry are always passed during deployment but will add a safety check anyway
Thanks!

Comment (sumithreddy) [reply objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:18]: done

Comment (SeemaTahaliyani) [reply objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:18]: can you explain this a bit please, is it possible for both of below to be true
.Values.global.storageClassName  .Values.atlas.persistence.storageClassName, how is storageClass set then?
`storageClassName: {{ default .Values.global.storageClassName .Values.atlas.persistence.storageClassName }}`

Comment (sumithreddy) [reply objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:18]: default <default val> <override val>

**Review by SeemaTahaliyani — APPROVED** [review]

**Comment by sumithreddy** [comment]
> please run custom-ci job and post the result before merging.

done and listed

File: objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml (hunk 1 above)
Comment (sumithreddy) [reply objectscale-iam/templates/atlas_v1beta1_iamcluster.yaml:18]: @eddiepavkovic made necessary changes. can you approve the PR?

**Review by eddiepavkovic — APPROVED** [review]
lgmt

**Pull request merged**